apiv2/project/project.go

```diff
@@ -468,13 +439,12 @@ func (c *RESTClient) UpdateProjectMetadata(ctx context.Context, p *modelv2.Proje
 		return &ErrProjectNotProvided{}
 	}
 
-	pID := int64(p.ProjectID)
 	metaKeyName := string(key)
 
 	_, err := c.LegacyClient.Products.GetProjectsProjectIDMetadatasMetaName(
 		&products.GetProjectsProjectIDMetadatasMetaNameParams{
 			MetaName:  metaKeyName,
-			ProjectID: pID,
+			ProjectID: int64(p.ProjectID),
```

Comment:
Maybe it would make more sense to change the type of `p.ProjectID` from` int32` to `int64`?

Reply:
Note that `GetProjectsProjectIDMetadatasMetaNameParams` originates from the `legacy` API and `p` originates from the `v2` API. 
In the `v2` API _many_ `project` API routes still use the `legacy` API, which is why we have convert the `ProjectID` in some places.